### DISCUSSION TITLE
I can't overwrite custom ssystem_minor.ini file in custom location

### DISCUSSION THREAD

**Discussion by heafnerj** [Stellarium/stellarium#3641] [Q&A] [answered]
I'm running Stellarium 23.4 on macOS 14.3.1 on my M1 iMac and my Intel MacBook Pro. I want to create a custom `ssystem_minor.ini` file containing only the objects I'm currently observing. I succeeded in deleting unneeded objects from  `~/Library/Application Support/Stellarium/data/ssystem_minor.ini` and of course I need to save the edited file in a custom location, and I chose `~/Documents/Stellarium/ssystem_minor.ini`. The problem is that if the latter file already exists, I cannot overwrite it despite confirming doing so in the save dialog. I have to manually delete the existing file and once again click "Export the Solar System minor objects to file..." button in the "Configuration file" tab of the Solar System Editor. Is this a bug or is it the intended behavior? I could not find an issue having filed on this particular matter so I decided to ask here. 

Also, should I be able to edit the file location in the Solar System Editor's "Configuration file" tab to point to my custom `ssystem_minor.ini` file?

**Comment by alex-w** [comment]
> I'm running Stellarium 23.4 on macOS 14.3.1 on my M1 iMac and my Intel MacBook Pro. I want to create a custom ssystem_minor.ini file containing only the objects I'm currently observing. I succeeded in deleting unneeded objects from  `~/Library/Application Support/Stellarium/data/ssystem_minor.ini`

OK

> and of course I need to save the edited file in a custom location, and I chose `~/Documents/Stellarium/ssystem_minor.ini`

Well, the answer is "Yes" here, if you want to save a backup version of custom file and answer is "No", if you want just to use custom file.

> The problem is that if the latter file already exists, I cannot overwrite it despite confirming doing so in the save dialog. I have to manually delete the existing file and once again click "Export the Solar System minor objects to file..." button in the "Configuration file" tab of the Solar System Editor. Is this a bug or is it the intended behavior? I could not find an issue having filed on this particular matter so I decided to ask here.

I can reproduce this behaviour in my M1 MBP. 

> Also, should I be able to edit the file location in the Solar System Editor's "Configuration file" tab to point to my custom `ssystem_minor.ini` file?

No, it's not possible to do this file separately from other configuration files.

**Reply by heafnerj** [reply]
Thank you.

**Comment by gzotti** [comment]
> The problem is that if the latter file already exists, I cannot overwrite it despite confirming doing so in the save dialog.

This is done on purpose. QFile::copy() does not overwrite. Of course, somebody could add some more logic, test for existence, show a confirmation dialog, ...

**Reply by alex-w** [reply]
I see a confirmation dialog and it rewriting the old file

**Reply by gzotti** [reply]
OK, right. There is a confirmation built in by default in QFileDialog::getSaveFileName() (unless you suppress it in the calling options).  The copy() function used after that does not overwrite and has no option to override. Someone would have to add the "if file exists then delete before calling copy()" logic. Looks trivial at first, however, it's more than two lines: do that only if file access permissions are OK!

**Reply by gzotti** [reply, marked as answer]
I have pushed a tentative solution. It works on Windows, but the overwrite behaviour on other platforms may vary.

**Reply by heafnerj** [reply]
I just tested it and it worked! Thank you very much.
